musicbot/bot.py

```diff
@@ -3880,19 +3886,22 @@ async def on_message(self, message):
 
         handler = getattr(self, "cmd_" + command, None)
         if not handler:
-            # alias handler
-            if self.config.usealias:
-                alias_command = self.aliases.get(command)
-                command, *alias_args = alias_command.split(" ")
-               
-                for alias_arg in alias_args:
-                    args.append(alias_arg)
-                
-                handler = getattr(self, "cmd_" + command, None)
-
-                if not handler:
-                    return
-            else:
+            modified_command = self.command_modifier.modifyUsingAlias(command)
+            modified_command, *modified_args = modified_command.split(" ")
+            
+            if modified_command == "gacha":
+                modified_command = self.command_modifier.modifyUsingGacha(" ".join(modified_args))
+                modified_command, *modified_args = modified_command.split(" ")
+            elif modified_command == "" and command == "gacha":
+                modified_command = self.command_modifier.modifyUsingGacha(" ".join(args))
+                modified_command, *modified_args = modified_command.split(" ")
```

Comment:
Suggestion: Process these inside ModifiedCommand, so we'll only need to replace `alias_command = self.aliases.get(command)` in the original code